plugins/woocommerce/includes/tracks/class-wc-tracks.php

```diff
@@ -67,6 +67,21 @@ public static function get_server_details() {
 		return $data;
 	}
 
+	/**
+	 * Get role-related details.
+	 *
+	 * @param WP_User $user The user object.
+	 * @return array The role details.
+	 */
+	public static function get_role_details( $user ) {
+		return array(
+			'role'                   => ! empty( $user->roles ) ? $user->roles[0] : '',
```

Comment:
Is this the default logic? Will there be a case where user can have multiple roles? If so, what order are they in? Unit tests around this would be good too, just to be clear about the desired logic.

Reply:
Adrian [found that](https://github.com/woocommerce/woocommerce/issues/52402) WordPress only provides a UI for giving a user a single role, but it is programmatically possible to give a user multiple roles.

We decided to keep it simple so that we only consider the first role in the list of roles for a user. I will add unit tests to cover this!

Reply:
Updated unit tests in 01bf835ec2c52d01e43c639047b2323131b62c4a.